sdk/lib/atl/atlbase.h

```diff
@@ -1837,14 +1937,53 @@ inline HRESULT WINAPI AtlComModuleRevokeClassObjects(_ATL_COM_MODULE *module)
 
     for (iter = module->m_ppAutoObjMapFirst; iter < module->m_ppAutoObjMapLast; iter++)
     {
-        hr = CoRevokeClassObject((*iter)->dwRegister);
+        _ATL_OBJMAP_ENTRY *ptr = *iter;
+        if (!ptr)
+            continue;
+
+        hr = CoRevokeClassObject(ptr->dwRegister);
         if (FAILED(hr))
             return hr;
     }
 
     return S_OK;
 }
 
+// Adapted from dll/win32/atl/atl.c
```

Comment:
Shouldn't these ATL dlls ideally be based on that sdk/lib/atl code base, instead of whatever is in win32/atl/ ?